src/main/java/com/configcat/ConfigCatHooks.java

```diff
@@ -116,11 +116,11 @@ void invokeOnClientReady(ClientCacheState clientCacheState) {
         }
     }
 
-    void invokeOnError(String error) {
+    void invokeOnError(Object error) {
         lock.readLock().lock();
         try {
             for (Consumer<String> func : this.onError) {
-                func.accept(error);
+                func.accept(error.toString());
```

Comment:
nit: There is a possible micro-optimization here: we could get the error message string before the for loop instead of calling `toString` for each consumer.